root.tf

```diff
@@ -271,11 +271,51 @@ module "ecr_draft_metadata_validator_repository" {
   source           = "./da-terraform-modules/ecr"
   repository_name  = "draft-metadata-validator"
   image_source_url = "https://github.com/nationalarchives/tdr-draft-metadata-validator/blob/main/Dockerfile"
-  allowed_principals = [
-    "arn:aws:iam::${data.aws_ssm_parameter.intg_account_number.value}:role/tdr-draft-metadata-validator-intg-role",
-    "arn:aws:iam::${data.aws_ssm_parameter.staging_account_number.value}:role/tdr-draft-metadata-validator-staging-role",
-    "arn:aws:iam::${data.aws_ssm_parameter.prod_account_number.value}:role/tdr-draft-metadata-validator-prod-role"
-  ]
+  repository_policy = jsonencode({
```

Comment:
This policy should be in a `json.tpl` and put in the `/templates/iam_policy` directory. It can then still be passed into the `da-terraform-module/ecr` module in the same way, ie via the `repository_policy` variable.

An example of passing in a full policy as a variable can be found here: https://github.com/nationalarchives/da-reference-generator/blob/77164b5812325170988d19e147ff372bfb609a92/terraform/root_main.tf#L32

So in this case it would look something like this:
```
...
repository_policy = templatefile("${path.module}/templates/iam_policy/{name of policy file}.json.tpl", {
      policy file param = param value
      ... etc ...
    })
...
```